src/caselawclient/factories.py

```diff
@@ -0,0 +1,168 @@
+import datetime
+from typing import Any, cast
+from unittest.mock import Mock, patch
+
+from typing_extensions import TypeAlias
+
+from caselawclient.Client import MarklogicApiClient
+from caselawclient.models.documents import Document
+from caselawclient.models.documents.body import DocumentBody
+from caselawclient.models.judgments import Judgment
+from caselawclient.models.press_summaries import PressSummary
+from caselawclient.responses.search_result import SearchResult, SearchResultMetadata
+
+
+class DocumentBodyFactory:
+    # "name_of_attribute": "default value"
+    PARAMS_MAP: dict[str, Any] = {
+        "name": "Judgment v Judgement",
+        "court": "Court of Testing",
+        "document_date_as_string": "2023-02-03",
+    }
+
+    @classmethod
+    def build(cls, **kwargs: Any) -> DocumentBody:
+        if "xml" in kwargs:
+            xml_string = kwargs.pop("xml")
+        else:
+            xml_string = "<akomantoso>This is some XML of a judgment.</akomantoso>"
+
+        document_body = DocumentBody(
+            xml_bytestring=xml_string.encode(encoding="utf-8"),
+        )
+
+        for param_name, default_value in cls.PARAMS_MAP.items():
+            if param_name in kwargs:
+                value = kwargs[param_name]
+            else:
+                value = default_value
+            setattr(document_body, param_name, value)
+
+        return document_body
+
+
+class DocumentFactory:
+    # "name_of_attribute": "default value"
+    PARAMS_MAP: dict[str, Any] = {
+        "is_published": False,
+        "is_sensitive": False,
+        "is_anonymised": False,
+        "is_failure": False,
+        "source_name": "Example Uploader",
+        "source_email": "uploader@example.com",
+        "consignment_reference": "TDR-12345",
+        "assigned_to": "",
+        "versions": [],
+    }
+
+    target_class: TypeAlias = Document
+
+    @classmethod
+    def build(cls, uri: str = "test/2023/123", **kwargs: Any) -> target_class:
```

Comment:
Does it make sense to have the explicitly named kwargs (html, body) as explicit optional parameters?